app/src/main/java/org/wikipedia/main/MainActivity.java

```diff
@@ -250,18 +251,23 @@ protected void clearToolbarElevation() {
 
     private class DrawerViewCallback implements MainDrawerView.Callback {
         @Override public void loginLogoutClick() {
-            if (AccountUtil.isLoggedIn()) {
-                WikipediaApp.getInstance().logOut();
-                FeedbackUtil.showMessage(MainActivity.this, R.string.toast_logout_complete);
-                if (Prefs.isReadingListSyncEnabled() && !ReadingListDbHelper.instance().isEmpty()) {
-                    ReadingListSyncBehaviorDialogs.removeExistingListsOnLogoutDialog(MainActivity.this);
-                }
-                Prefs.setReadingListsLastSyncTime(null);
-                Prefs.setReadingListSyncEnabled(false);
-            } else {
-                getFragment().onLoginRequested();
-            }
-            closeMainDrawer();
+            new AlertDialog.Builder(MainActivity.this)
+                    .setMessage(R.string.logout_prompt)
+                    .setNegativeButton(android.R.string.cancel, null)
+                    .setPositiveButton(R.string.preference_title_logout, (dialog, which) -> {
+                        if (AccountUtil.isLoggedIn()) {
+                            WikipediaApp.getInstance().logOut();
+                            FeedbackUtil.showMessage(MainActivity.this, R.string.toast_logout_complete);
+                            if (Prefs.isReadingListSyncEnabled() && !ReadingListDbHelper.instance().isEmpty()) {
+                                ReadingListSyncBehaviorDialogs.removeExistingListsOnLogoutDialog(MainActivity.this);
+                            }
+                            Prefs.setReadingListsLastSyncTime(null);
+                            Prefs.setReadingListSyncEnabled(false);
+                        } else {
+                            getFragment().onLoginRequested();
```

Comment:
The `AccountUtil.isLoggedIn()` should be moved out as the main logic of behavior, otherwise, users will not be able to log in.

Reply:
🤦‍♀️ doh, that is correct!